dcos_launch/__init__.py

```diff
@@ -16,4 +17,9 @@ def get_launcher(config):
             return dcos_launch.onprem.OnpremLauncher(config)
     if platform == 'azure':
         return dcos_launch.arm.AzureResourceGroupLauncher(config)
+    if platform == 'gce':
+        if provider == 'onprem':
+            return dcos_launch.onprem.OnpremLauncher(config)
+        else:
+            raise dcos_launch.util.LauncherError('UnsupportedAction', 'Launch provider {} not supported for the {} platform.'.format(provider, platform))
```

Comment:
@mellenburg 
1. why not just leave this as a fall through?
A: The error message is slightly different from the one below
2. No need for the else
A: +1